### DISCUSSION TITLE
Problem opening a AlertDialog

### DISCUSSION THREAD

**Discussion by R8bert** [flet-dev/flet#1871] [Q&A] [answered]
HI, i have a problem creating a AlertDialog after one is closed, i went to github and integrated a calendar for my app and now i have a problem with not fully understanting flet or beeing a pro at python

this is the full code 

```
    def __init__(self):
        super().__init__()
       
        self.cal_ico_xx1 = IconButton(
            icon=icons.CALENDAR_MONTH, 
            on_click=self.open_dlg_modalXX1
            )

        self.st = Stack(
            [
                self.tf,
                self.cal_ico,
            ]
        )
        self.st1 = Stack(
            [
                self.cal_ico_xx1,
            ]
        )
        self.dlg_modalXX1 = AlertDialog(
            modal=True,
            title=Text("Date picker"),
            actions=[
                TextButton("Cancel", on_click=self.cancel_dlg_xx1),
                TextButton("Confirm", on_click=self.confirm_dlg_xx1),
            ],
            actions_alignment=MainAxisAlignment.END,
            actions_padding=5,
            content_padding=0
        )
        self.dlg_modalXX = AlertDialog(
            title=Text("Date picker"),
            actions=[
                TextButton("Cancel", on_click=self.cancel_dlg),
                TextButton("Confirm", on_click=self.confirm_dlg,),
                self.st1
            ],
            actions_alignment=MainAxisAlignment.END,
            actions_padding=5,
            content_padding=0
        )
    def confirm_dlg(self,e):
        self.dlg_modalXX.open = False
        self.dlg_modalXX.update()

        print("not an error")

    def confirm_dlg_xx1(self,e):
        self.dlg_modalXX1.open = False
        self.dlg_modalXX1.update()

        print("not an error")


        # x = self.machine.value
        # testlinkdinamic(x)
        
        
        # self.page.update()
    
    def cancel_dlg(self,e):
        self.dlg_modalXX.open = False
        self.dlg_modalXX.update()
    

    def cancel_dlg_xx1(self,e):
        self.dlg_modalXX1.open = False
        self.dlg_modalXX1.update()


    def open_dlg_modalXX(self, e):
        self.datepicker = DatePicker(
            hour_minute=self.c1.value,
            selected_date=[self.tf.value] if self.tf.value else None,
            selection_type=int(self.cg.value),
            disable_to=self._to_datetime(self.tf1.value),
            disable_from=self._to_datetime(self.tf2.value),
            hide_prev_next_month_days=self.c2.value,
            holidays=self.holidays,
            show_three_months=self.c3.value,
            on_change=self.callback
            )
        self.page.dialog = self.dlg_modalXX
        self.dlg_modalXX.content = self.datepicker
        self.dlg_modalXX.open = True
        self.page.update()

    def open_dlg_modalXX1(self, e):
        self.page.dialog = self.dlg_modalXX1
        self.contxx1 = TextField(label="Disable days until date")
        self.dlg_modalXX1.content = self.contxx1
        self.dlg_modalXX1.open = True
        self.page.update()
```

i have problems there:

```
 def open_dlg_modalXX(self, e):
        self.datepicker = DatePicker(
            hour_minute=self.c1.value,
            selected_date=[self.tf.value] if self.tf.value else None,
            selection_type=int(self.cg.value),
            disable_to=self._to_datetime(self.tf1.value),
            disable_from=self._to_datetime(self.tf2.value),
            hide_prev_next_month_days=self.c2.value,
            holidays=self.holidays,
            show_three_months=self.c3.value,
            on_change=self.callback
            )
        self.page.dialog = self.dlg_modalXX
        self.dlg_modalXX.content = self.datepicker
        self.dlg_modalXX.open = True
        self.page.update()

    def open_dlg_modalXX1(self, e):
        self.page.dialog = self.dlg_modalXX1
        self.contxx1 = TextField(label="Disable days until date")
        self.dlg_modalXX1.content = self.contxx1
        self.dlg_modalXX1.open = True
        self.page.update()
```
        
if i try to open the modalXX1 in modalXX i get AttributeError: 'NoneType' object has no attribute 'dialog'

Some help please, i struggle for 2 days on this and i dont understand what i am doing wrong

**Comment by mouadessalim** [comment]
Firstly, the issue description appears to be quite complex and challenging to understand. To facilitate a swift resolution, could you please provide a more concise and clear explanation of the problem you're facing? Specific details about the issue and steps to reproduce it would be extremely helpful.

Secondly, instead of copying and pasting the entire program into the issue, it would be more efficient to pinpoint the specific section or lines of code where you believe the issue is occurring. This focused approach not only helps us identify the problem area quickly but also keeps the discussion more organized.

We value professionalism and efficiency in this open-source collaboration, and your assistance in this matter will contribute to a more productive discussion. If you have any questions or need guidance on how to create a more effective issue, please don't hesitate to ask.

**Reply by R8bert** [reply]
i updated the post,sorry

**Comment by ndonkoHenri** [comment]
Please provide some runnable code, so we can try to help you out...

Nevertheless, I guess the problem is here: `self.page.dialog = self.dlg_modalXX`
`self.page` is `None` at this level, hence the error...

**Reply by R8bert** [reply, marked as answer]
it connects to a local offline database so you cant run it, i can explain it tho
![image](https://github.com/flet-dev/flet/assets/61593351/8aafb368-a92a-462f-8e74-ef0c0a16cc20)
this is a dashboard card who opens a alertdialog with the content from Example
![image](https://github.com/flet-dev/flet/assets/61593351/dd319786-8141-47aa-be10-7a58bee03661)
![image](https://github.com/flet-dev/flet/assets/61593351/2c443ea1-64aa-45c1-9f82-a4282435446f)
![image](https://github.com/flet-dev/flet/assets/61593351/1c6ec3d2-df5f-4fcd-a627-3240a893a8c2)
and this calendar icon opens this picker 
who contains a icon too but when i click it i get `'NoneType' object has no attribute 'dialog' ` but this is for self.dlg_modalXX1, the dialog for self.dlg_modalXX works fine
if i try to run just the calendar without the dashboard card selector thing both dialogs are working

**Comment by R8bert** [comment]
I just created another page with the AlertDialog content, and now it's working well. I will just add a sidebar and call it a day
